File: lib/models/course.dart
class Course {
  final String id;
  final String title;
  final String description;

  Course({
    required this.id,
    required this.title,
    required this.description,
  });

  factory Course.fromJson(Map<String, dynamic> json) {
    return Course(
      id: json['_id'] ?? json['id'],
      title: json['title'],
      description: json['description'] ?? '',
    );
  }
}

File: backend/src/models/Course.js
import mongoose from "mongoose";

const courseSchema = new mongoose.Schema(
  {
    title: {
      type: String,
      required: true,
    },

    description: {
      type: String,
    },

    // 🔥 STRING because auth uses "lect1"
    lecturerId: {
      type: String,
      required: true,
    },
  },
  { timestamps: true }
);

// ✅ Prevent OverwriteModelError
export default mongoose.models.Course ||
  mongoose.model("Course", courseSchema);
